client/src/api/block_builder/input_selection/core/requirement/amount.rs

```diff
@@ -147,6 +150,14 @@ impl AmountSelection {
                 *self.inputs_sdr.entry(*sdruc.return_address()).or_default() += sdruc.amount();
             }
 
+            if !input.output.is_basic() {
```

Comment:
What about aliases with governance transition? Then we can't do this, no?

Reply:
This is only evaluated on available inputs so if a governance transition was already provided in output or required because of a sender feature, by the time of amount selection it wouldn't be available anymore. And for amount we give a None (L162) as transition which will default to State